example/vm/config.yaml

```diff
@@ -134,14 +134,17 @@ processors:
         # For more details, see:
         # https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor
         actions:
-            # In this case, the attributes processor maps the `service_version` attribute attached
-            # to telemetry from the prometheus/self receiver to `service.version` following OTel
-            # semantic conventions for service attributes.
+            # In this case, the attributes processor maps service attributes attached to telemetry
+            # from the prometheus/self receiver to service attributes following OTel semantic
+            # conventions for service attributes.
             # For more details, see:
             # https://opentelemetry.io/docs/specs/semconv/attributes-registry/service/
             - key: service.version
               action: insert
               from_value: service_version
+            - key: service.version
+              action: upsert
```

Comment:
what are the expected outputs here for different combinations of the existence of `service_version` and `collector_name` on a given metric?

like if both are present, do we let `collector_name` take precedence? and otherwise its just whichever field is present?

Reply:
🤦 I messed up, this second one was supposed to be `service.name`

Here's the guide for how the attribute processing actions work: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/attributesprocessor#attributes-processor

* **insert** for `service_version` to `service.version` creates `service.version` if it doesn't already exist
* **upsert** for `collector_name` to `service.name` creates or replaces `service.name`